obal/data/modules/koji_verify_tag.py

```diff
@@ -0,0 +1,64 @@
+"""
+Verify packages against a tag in Koji
+"""
+
+import os
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.obal import get_specfile_name # pylint:disable=import-error,no-name-in-module
+from ansible.module_utils.koji_wrapper import koji # pylint:disable=import-error,no-name-in-module
+
+def main():
+    """
+    Verify a tag
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            packages=dict(type='dict', required=True),
+            tag=dict(type='str', required=True),
+            directory=dict(type='str', required=True),
+            koji_executable=dict(type='str', require=False, default='koji')
+        )
+    )
+
+    packages_for_tag = set()
+
+    for (package, attributes) in module.params['packages'].items():
+        tag = None
+
+        if 'koji_tags' in attributes:
+            for koji_tag in attributes['koji_tags']:
+                if module.params['tag'] == koji_tag['name']:
+                    tag = koji_tag
+
+            if tag and 'package_base_dir' in attributes:
+                specfile = os.path.join(attributes['package_base_dir'], package, "{}.spec".format(package))
+                scl = tag.get('scl')
+                name = get_specfile_name(os.path.join(module.params['directory'], specfile), scl)
+                packages_for_tag.add(name)
+
+    try:
+        command = ['list-pkgs', '--quiet', '--tag', module.params['tag']]
+        koji_output = koji(
+            command,
+            executable=module.params['koji_executable']
+        )
+
+        packages_in_koji = {item.split(' ', 1)[0] for item in koji_output.split("\n") if item}
+
+    except subprocess.CalledProcessError as error:
+        module.fail_json(changed=False, msg=error.output)
+
+    missing_in_koji = packages_for_tag - packages_in_koji
+    missing_in_git = packages_in_koji - packages_for_tag
+
+    module.exit_json(
```

Comment:
Can this fail `if missing_in_koji or missing_in_git`? Ansible has a way to continue on error, right? Or perhaps the regular failure output is sufficient anyway.

Reply:
Yes there are two options:

 1) Have the module fail and allow users of the module to ignore errors if they want
 2) Have this module simply report data and then have a task that fails conditionally

I do not know what is more correct, so I am Ok with either strategically.

Reply:
I'd lean to option 1.